packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,33 +1,292 @@
+          currentMenuItems = [...currentMenuItems, itemToHide]
+          currentMenuItemsWidths = [...currentMenuItemsWidths, itemToHideWidth]
+          currentVisibleItems = [...currentVisibleItems.slice(1)]
+          currentVisibleItemWidths = [...currentVisibleItemWidths.slice(1)]
+
+          visibleItemsWidthTotal = calculateVisibleItemsWidth(currentVisibleItemWidths)
+
+          // Add menu button width
+          if (currentMenuItems.length > 0) {
+            visibleItemsWidthTotal += MENU_BUTTON_WIDTH
+          }
+
+          // If hideRoot is false but we still don't fit with root + menu + leaf,
+          // fallback to hideRoot=true behavior (menu + leaf only)
+          if (!hideRoot && currentVisibleItems.length === 1 && visibleItemsWidthTotal > availableWidth) {
+            eHideRoot = true
+            break
+          } else {
+            eHideRoot = hideRoot
+          }
+        }
+      }
+      return {
+        visibleItems: [...currentVisibleItems],
+        menuItems: [...currentMenuItems],
+        effectiveHideRoot: eHideRoot,
+      }
+    },
+    [
+      MIN_VISIBLE_ITEMS,
+      childArray,
+      childArrayWidths,
+      effectiveHideRoot,
+      hideRoot,
+      overflow,
+      rootItemWidth,
+      menuButtonWidth,
+    ],
+  )
+
+  const handleResize = useCallback(
+    (entries: ResizeObserverEntry[]) => {
+      if (entries[0]) {
+        const containerWidth = entries[0].contentRect.width
+        const result = calculateOverflow(containerWidth)
+        setVisibleItems(result.visibleItems)
+        setMenuItems(result.menuItems)
+        setEffectiveHideRoot(result.effectiveHideRoot)
+      }
+    },
+    [calculateOverflow],
+  )
+
+  useResizeObserver(handleResize, containerRef)
+
+  // Initial overflow calculation for testing and >5 items
+  useEffect(() => {
+    if (overflow === 'menu' && childArray.length > 5) {
+      // Get actual container width from DOM or use default
+      const containerWidth = containerRef.current?.offsetWidth || 800
+      const result = calculateOverflow(containerWidth)
+      setVisibleItems(result.visibleItems)
+      setMenuItems(result.menuItems)
+      setEffectiveHideRoot(result.effectiveHideRoot)
+    }
+  }, [overflow, childArray.length, calculateOverflow])
+
+  // Determine final children to render
+  const finalChildren = React.useMemo(() => {
+    if (overflow === 'wrap' || menuItems.length === 0) {
+      return React.Children.map(children, child => <li className={classes.ItemWrapper}>{child}</li>)
+    }
+
+    let effectiveMenuItems = [...menuItems]
+    if (!effectiveHideRoot) {
+      effectiveMenuItems = [...menuItems.slice(1)]
+    }
+    const menuElement = (
+      <li className={classes.BreadcrumbsItem} key="breadcrumbs-menu">
+        <BreadcrumbsMenuItem
+          ref={menuButtonRef}
+          items={effectiveMenuItems}
+          aria-label={`${effectiveMenuItems.length} more breadcrumb items`}
```

Comment:
This could be a good use for the `…` character specifically on toggle element (got the idea from spectrum: https://react-spectrum.adobe.com/react-spectrum/Breadcrumbs.html) and that might mean we don't need a tooltip on this